RELEASE.md

```diff
@@ -1,33 +1,65 @@
-To release a new version of qtpy on PyPI:
+# Release Procedure
 
-* Close Github milestone
+To release a new version of QtPy on PyPI:
 
-* git fetch upstream && git merge upstream/master
 
-* git clean -xfdi
+## Prepare
 
-* Update CHANGELOG.md using `loghub` to generate the list of issues and PRs merged to add at the top of the file
+* Close Github milestone and ensure all issues are resolved/moved
 
-    loghub -m vX.X.X spyder-ide/qtpy
+* `git switch <release-branch>` then `hub sync` or `git fetch upstream && git merge upstream/<release-branch>`
 
-* Update `_version.py` (set release version, remove 'dev0')
+* Check `MANIFEST.in` and `setup.cfg` to ensure they are up to date
 
-* git add . && git commit -m 'Release X.X.X'
+* `git clean -xfdi`
 
-* python setup.py sdist
 
-* python setup.py bdist_wheel
+## Commit
 
-* twine check dist/*
+* Update `CHANGELOG.md` using `loghub` to generate the list of issues and PRs merged to add at the top of the file:
 
-* twine upload dist/*
+  ```bash
+  loghub -m vX.X.X spyder-ide/qtpy
+  ```
 
-* git tag -a vX.X.X -m 'Release X.X.X'
+* Update `__version__` in `__init__.py` (set release version, remove `.dev0`)
 
-* Update `_version.py` (add 'dev0' and increment minor)
+* `git commit -am 'Release X.X.X'`
 
-* git add . && git commit -m 'Back to work'
 
-* git push
+## Build
 
-* git push --tags
+* `conda install build packaging twine` or `pip install build packaging twine` (if not installed)
+
+* `conda update build packaging pip setuptools twine wheel` or `pip install --upgrade build packaging pep517 pip setuptools twine wheel`
+
+* `python -bb -X dev -W error -m build`
+
+
+## Test
+
+* `python -bb -X dev -W error -m pip install dist/*.whl`
+
+* `pytest qtpy/tests`
+
+* `python -bb -X dev -W error -m pip install -e .`
+
+
+## Upload
+
+* `twine check --strict dist/*`
+
+* `twine upload dist/*`
+
+* `git tag -a vX.Y.Z -m 'Release X.Y.Z'`
```

Comment:
I modified this because `X.X.X` implies that the the version components are the same variable repeated (e.g. 1.1.1), whereas `X.Y.Z` makes it clear that they are three different variables, and that the two different uses of the X.Y.Z string _are_ equivalent. Furthermore, this is the most widely used standard convention (at least in English) for describing arbitrary version and tag names, e.g. in [SemVer](https://semver.org/) itself, and a search for `version "X.Y.Z"` gives 400 million hits for mostly version-related things, while `version "X.X.X"` gives you 600 million hits that are overwhelmingly dominated by...well, something that isn't very family-friendly (one more reason to avoid it, heh).

Reply:
Then we should use the same convention in the other commands above (they still are using the `X.X.X` convention)

Reply:
Ah, great point—don't know how I missed those. I went back and did so, thanks.